File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_map>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    unordered_map<int, bool> m;
    int pType = 0;
    for (int i = 0; i < nums.size(); i++) {
        if (m.find(nums[i]) == m.end()) {
            m[nums[i]] = true;
            pType++;
        }
    }
    if (pType >= nums.size() / 2) answer = nums.size() / 2;
    else answer = pType;
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        
        for(int num: nums) {
            set.add(num);
        }
        
        int maxPick = nums.length / 2;
        int typeCount = set.size();
        
        return Math.min(maxPick, typeCount);
    }
}